### PR TITLE
[USB] Change wheel type to device subtype

### PR DESCRIPTION
**Pull request by Florin9doi**
It is used only for Steering Wheels now, but may allows later to group more devices like Keyboard/Beatmania, Microphone/Singstar, or RealPlay.

### CODE REVIEW

**Comment by refractionpcsx2** [comment]
Looks like Linux isn't happy here

File: pcsx2/USB/usb-pad/raw/usb-pad-raw.cpp [hunk 1]
```diff
@@ -253,7 +253,7 @@ namespace usb_pad
 					uint16_t btn = mapping->btnMap[usage[i] - pButtonCaps->Range.UsageMin];
 					for (int j = 0; j < 2; j++)
 					{
-						PS2WheelTypes wt = (PS2WheelTypes)conf.WheelType[1 - j];
+						PS2WheelTypes wt = (PS2WheelTypes) 0; // conf.WheelType[1 - j];
```
Comment (jackun) [inline pcsx2/USB/usb-pad/raw/usb-pad-raw.cpp:256]: Pass type from RawInputPad::ParseRawInput?

Comment (jackun) [inline pcsx2/USB/usb-pad/raw/usb-pad-raw.cpp:256]: Pass type from RawInputPad::ParseRawInput?

**Comment by jackun** [comment]
Thought of "pulling" the device subtypes "upward" if it got any like Rock band drum kit and Buzz. Any point?

**Comment by Florin9doi** [comment]
You mean adding BUZZ wired and wireless emulation? Unless they make any difference for games, I think it would add unnecessary worries for users.

**Comment by jackun** [comment]
I mean, only have the single device combo box. Instead of "Wheel device", you have directly "Driving Force, DF Pro, ...blah..., Singstar, ..blah.., Keyboardmania" etc.

**Comment by Florin9doi** [comment]
Having a single combo box would result in a really long list. There are still device to be added: lightguns, realplay, gametrak, dj hero, psp.

**Comment by RedDevilus** [comment]
What about just having 2 selections? 1 side just gives context menu for Device types and then on right side the list of devices within your first selection of the device type.

File: pcsx2/USB/Win32/Config_usb.cpp [hunk 2]
```diff
@@ -64,6 +64,18 @@ void SelChangedAPI(HWND hW, int port)
 	changedAPIs[std::make_pair(port, devName)] = *it;
 }
 
+void SelChangedSubtype(HWND hW, int port)
+{
+	int sel = SendDlgItemMessage(hW, port ? IDC_COMBO_WHEEL_TYPE1_USB : IDC_COMBO_WHEEL_TYPE2_USB, CB_GETCURSEL, 0, 0);
```
Comment (lightningterror) [inline pcsx2/USB/Win32/Config_usb.cpp:69]: You can use const here, didn't check others but would be nice to change them too if there are any that can be changed.

**Comment by jackun** [comment]
Seems good but return std::vector for static strings seems weird and I can't think a better way than some gls::span or c++20 thingy.

File: pcsx2/USB/usb-pad/usb-pad.h [hunk 3]
```diff
@@ -59,6 +59,15 @@ namespace usb_pad
 		static const TCHAR* LongAPIName(const std::string& name);
 		static int Configure(int port, const std::string& api, void* data);
 		static int Freeze(int mode, USBDevice* dev, void* data);
+		static std::vector<std::string> SubTypes()
+		{
+			std::vector<std::string> ret;
```
Comment (jackun) [inline pcsx2/USB/usb-pad/usb-pad.h:64]: This probably could just be a initializer list: `{ "blahh", }` etc

Comment (Florin9doi) [reply pcsx2/USB/usb-pad/usb-pad.h]: done

**Review by jackun — APPROVED** [review]

**Comment by Florin9doi** [comment]
anything else needed ?

**Pull request merged**